pkg/api/v1/atlasproject_types.go

```diff
@@ -156,8 +156,12 @@ func (p *AtlasProject) UpdateStatus(conditions []status.Condition, options ...st
 }
 
 func (p *AtlasProject) X509SecretObjectKey() *client.ObjectKey {
+	namespace := p.Spec.X509CertRef.Namespace
```

Comment:
Hey, thanks for the PR 😌
Can you please move this new code you've added to the inside of `if p.Spec.X509CertRef != nil`?

Reply:
Thank you 🙏